src/Symfony/Component/Filesystem/Filesystem.php

```
@@ -317,6 +317,76 @@ public function symlink($originDir, $targetDir, $copyOnWindows = false)
     }
 
     /**
+     * Return the next direct target of a link.
+     *
+     * Tto recursively follow links until a final target
+     * is reached, use @see Filesystem::realpath($path).
+     *
+     * @param string $path A link path.
+     *
+     * @return string Return the resolved link.
+     *
+     * @throws IOException When the link does not exist or is not readable.
+     */
+    public function readlink($path)
+    {
+        if (!$this->exists($path)) {
+            throw new IOException(sprintf('The link %s does not exist and cannot be read.', $path));
+        }
+
+        if (!is_link($path)) {
+            return $path;
```

PHP's `readlink()` function will emit a warning if the given path is no link. I think we should throw an `InvalidArgumentException` then.